src/main/java/marquez/client/models/DatasetMeta.java

```diff
@@ -14,26 +14,38 @@
 
 package marquez.client.models;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import java.util.Optional;
 import javax.annotation.Nullable;
-import lombok.Builder;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.NonNull;
-import lombok.Value;
-import marquez.client.utils.JsonUtils;
+import lombok.ToString;
 
-@Value
-@Builder
-public class DatasetMeta {
+@AllArgsConstructor(onConstructor = @__(@JsonCreator))
+@EqualsAndHashCode
+@ToString
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.EXISTING_PROPERTY,
+    property = "type")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = DbTableMeta.class, name = "DB_TABLE"),
+  @JsonSubTypes.Type(value = StreamMeta.class, name = "STREAM")
+})
+public abstract class DatasetMeta {
+  @Getter @NonNull private DatasetType type;
```

Comment:
_Minor_: Let's make each instance variable `private` and `final`.

Reply:
👍